### PR TITLE
Add focal loss without logits

### PR DESCRIPTION
**Pull request by ori-kishony**

### CODE REVIEW

**Comment by datumbox** [comment]
@ori-kishony Thanks for the contribution.

Unfortunately we won't be able to accept this PR because it adds a significant amount of duplicate code in the project. The majority of the added code comes from `sigmoid_focal_loss`. 

To maximize the chances of getting your PRs merged on the future and reduce throwaway work, please consider opening an issue and discussing the changes prior making code changes. We have a [guide](https://github.com/pytorch/vision/blob/main/CONTRIBUTING.md) on how to contribute to the project, please have a look. Thanks!
